tools/topology/topology2/platform/intel/sdw-amp-generic.conf

```diff
@@ -413,16 +413,22 @@ IncludeByKey.SDW_AMP_FEEDBACK {
 						# 32-bit 48KHz 2ch
 						Object.Base.input_audio_format [
 							{
+								in_channels		2
 								in_bit_depth            32
 								in_valid_bit_depth      $SDW_LINK_VALID_BITS
+								in_ch_cfg		$CHANNEL_CONFIG_STEREO
+								in_ch_map		$CHANNEL_MAP_STEREO
```

Comment:
Aren't they the default values?
See https://github.com/thesofproject/sof/blob/main/tools/topology/topology2/include/common/input_audio_format.conf#L128
and https://github.com/thesofproject/sof/blob/main/tools/topology/topology2/include/common/output_audio_format.conf#L125

@ranj063 Any idea?

Reply:
Well, it's not working unless those values are given explicitly.

Anyway, why rely on defaults if we can configure it? The comment says:

    # 32-bit 48KHz 2ch

So why not explicitly set 2ch ?

Reply:
Yes, I have no objection. Just curious why the default values don't work.

Reply:
Agree, if it should default to this config it would be good to understand why it isn't working. Maybe something else is breaking the default?

Reply:
I think I've found the actual problem is that the capture pipeline contains a mix of configurable channels and hardcoded channels. Specifically AMP_FEEDBACK_CH is configurable but for 1 link the only value that works is AMP_FEEDBACK_CH=2 because the rest of the pipeline is hardcoded to 2.

I have been working on a larger patch that fixes the capture pipeline to remove the hardcoded channel assumptions. I'll finish that and do a pull request for that one.